app/models/invoice.rb

```diff
@@ -268,7 +268,14 @@ def set_fields_from_stripe_invoice(inv = remote_invoice)
     self.auto_advance = inv.auto_advance
     self.due_date = Time.at(inv.due_date).to_datetime # convert from unixtime
     self.ending_balance = inv.ending_balance
-    self.finalized_at = inv.respond_to?(:status_transitions) ? inv.status_transitions.finalized_at : inv.try(:finalized_at)
+
+    finalized_value = if inv.respond_to?(:status_transitions)
+                        inv.status_transitions.finalized_at
+                      else
+                        inv.try(:finalized_at)
+                      end
+    self.finalized_at = finalized_value ? Time.at(finalized_value.to_i).to_datetime : nil
```

Comment:
According to these two issues below, the Rails 7 behavior was to silently set it to `nil` and it was never being assigned correctly. ActiveRecord 8 behavior is to raise a `PG::DatetimeFieldOverflow` error.

- https://github.com/rails/rails/issues/53679
- https://github.com/rails/rails/issues/54110

```ruby
ActiveRecord::StatementInvalid:
 PG::DatetimeFieldOverflow: ERROR: date/time field value out of range: "1752413473"
 HINT: Perhaps you need a different "datestyle" setting.
 CONTEXT: unnamed portal parameter $14 = '...'
```

I'm not sure if there are other occurrences of this that aren't covered by tests; If there are it might not be a bad idea to monkey patch it like in: https://github.com/rails/rails/issues/53679#issuecomment-2530827971

Reply:
not really sure why we need `respond_to?` when `status_transitions` isn't nullable: https://docs.stripe.com/api/invoices/object?api-version=2024-06-20#invoice_object-status_transitions
also doesn't look like the Invoice object can ever have a `finalized_at` field.

Reply:
extracted this fix to https://github.com/hackclub/hcb/pull/11872